full-stack-tests/presentation/src/backend/PresentationManager.test.ts

```diff
@@ -0,0 +1,130 @@
+        type:"Decimal",
+        uomSeparator:""};
+
+      const map = {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        AREA: {unitSystems: [PresentationUnitSystem.BritishImperial], format: formatProps},
+      };
+
+      const manager: PresentationManager = new PresentationManager({defaultFormats: map});
+      const descriptor = await manager.getContentDescriptor({
+        imodel,
+        rulesetOrId: ruleset,
+        keys,
+        displayType: "Grid",
+        requestContext: new ClientRequestContext(),
+        unitSystem: PresentationUnitSystem.BritishImperial,
+      });
+      expect(descriptor).to.not.be.undefined;
+      const field = findFieldByLabel(descriptor!.fields, "cm2")!;
+      expect(field).not.to.be.undefined;
+      const content = await manager.getContent({ imodel, rulesetOrId: ruleset, keys, descriptor: descriptor!, requestContext: new ClientRequestContext(), unitSystem: PresentationUnitSystem.BritishImperial });
+      const displayValues = content!.contentSet[0].displayValues.rc_generic_PhysicalObject_ncc_MyProp_areaElementAspect as DisplayValuesArray;
+      expect(displayValues.length).is.eq(1);
+      expect(((displayValues[0] as DisplayValuesMap).displayValues as DisplayValuesMap)[field.name]!).to.eq("0.1616 ft²");
+    });
+
+    it("formats property using provided format when it has provided format and default format for requested unit system", async () => {
+      const formatProps = {
+        composite:
+        {includeZero:true,
+          spacer:" ",
+          units:[{label:"ft²",name:"SQ_FT"}],
+        },
+        formatTraits:"KeepSingleZero|KeepDecimalPoint|ShowUnitLabel",
+        precision:4,
+        type:"Decimal",
+        uomSeparator:""};
+
+      const map = {
+        // eslint-disable-next-line @typescript-eslint/naming-convention
+        AREA: {unitSystems: [PresentationUnitSystem.Metric], format: formatProps},
+      };
+
+      const manager: PresentationManager = new PresentationManager({defaultFormats: map});
+      const descriptor = await manager.getContentDescriptor({
+        imodel,
+        rulesetOrId: ruleset,
+        keys,
+        displayType: "Grid",
+        requestContext: new ClientRequestContext(),
+        unitSystem: PresentationUnitSystem.Metric,
+      });
+      expect(descriptor).to.not.be.undefined;
+      const field = findFieldByLabel(descriptor!.fields, "cm2")!;
+      expect(field).not.to.be.undefined;
+      const content = await manager.getContent({ imodel, rulesetOrId: ruleset, keys, descriptor: descriptor!, requestContext: new ClientRequestContext(), unitSystem: PresentationUnitSystem.Metric });
+      const displayValues = content!.contentSet[0].displayValues.rc_generic_PhysicalObject_ncc_MyProp_areaElementAspect as DisplayValuesArray;
+      expect(displayValues.length).is.eq(1);
+      expect(((displayValues[0] as DisplayValuesMap).displayValues as DisplayValuesMap)[field.name]!).to.eq("150.1235 cm²");
```

Comment:
pretty much the same piece of code is used in all three tests - could we move it to a function?

Reply:
Sure, moved it to a separate function